.github/actions/test-setup/action.yml

```diff
@@ -1,56 +1,99 @@
   using: "composite"
   steps:
+    - name: Create directories for nodes
+      run: |
+        mkdir protected_node open_node
+      shell: bash
+
     # Note: Calling workflows will likely also require the checkout action,
     # but we include it here as an extra precaution so that the action can be run independently
     # since subsequent steps depend on the checkout action
-    - name: Checkout
+    - name: Checkout recipes for Protected node (full_stack node)
       uses: actions/checkout@v4
-    
-    - name: Rename files
+      path: protected_node/recipes
+
+    # Note: OpenNeuro is public data, but we use it for an example protected node here
+    # because Neurobagel synthetic data contains instances of all variables and is thus more flexible
+    # for tests involving an open node.
+    - name: Get data source for Protected node
+      uses: actions/checkout@v4
+      with:
+        repository: neurobagel/openneuro-annotations
+        path: protected_node/openneuro-annotations
+
+    - name: Get data for Protected node
       run: |
-        cp template.env .env
-        cp local_nb_nodes.template.json local_nb_nodes.json
+        mkdir protected_node/openneuro_mini_data
+  
+        # Note: Both ds000001 and ds000002 have only sex and age annotations
+        cp protected_node/openneuro-annotations/ds000001.jsonld protected_node/openneuro-annotations/ds000002.jsonld protected_node/openneuro_mini_data
       shell: bash
-    
-    - name: Edit environment variables and credentials
+
+    - name: Configure Protected node
+      working-directory: protected_node/recipes
       run: |
-        # Replace NB_API_QUERY_URL placeholder
-        sed -i 's|^NB_API_QUERY_URL=.*|NB_API_QUERY_URL=http://localhost:8080|g' .env
-        sed -i 's|^NB_FEDERATE_REMOTE_PUBLIC_NODES=.*|NB_FEDERATE_REMOTE_PUBLIC_NODES=false|g' .env
+        cp template.env .env
+        cp local_nb_nodes.template.json local_nb_nodes.json
+
+        # Replace defaults, even if they are commented out
+        sed -i 's|^#\?[[:space:]]\?COMPOSE_PROJECT_NAME=.*|COMPOSE_PROJECT_NAME=protected_node|' .env
+        sed -i 's|^#\?[[:space:]]\?LOCAL_GRAPH_DATA=.*|LOCAL_GRAPH_DATA=../openneuro_mini_data|' .env
+        sed -i 's|^#\?[[:space:]]\?NB_FEDERATE_REMOTE_PUBLIC_NODES=.*|NB_FEDERATE_REMOTE_PUBLIC_NODES=false|' .env
+        sed -i 's|^#\?[[:space:]]\?NB_API_QUERY_URL=.*|NB_API_QUERY_URL=http://localhost:8080|' .env
 
         # Replace default username/password
-        echo "testadminpassword" > secrets/NB_GRAPH_ADMIN_PASSWORD.txt
-        echo "testpassword" > secrets/NB_GRAPH_PASSWORD.txt
+        echo "testadminpassword1" > secrets/NB_GRAPH_ADMIN_PASSWORD.txt
+        echo "testpassword1" > secrets/NB_GRAPH_PASSWORD.txt
+
+        # Define local nodes to federate over
+        echo '[{"NodeName": "2 OpenNeuro Datasets", "ApiURL": "http://protected_node-api-1:8000"},{"NodeName": "BIDS Synthetic", "ApiURL": "http://open_node-api-1:8000"}]' > local_nb_nodes.json
```

Comment:
why not call the nodes "protected" and "open" here?

Reply:
So we can quickly glean from the UI (without having to download results) and from `local_nb_nodes.json` what data is in each node. This way we also replicate our recommended deployment workflow, which includes using informative node names :)